eval.c

```diff
@@ -399,6 +399,18 @@ static sexp sexp_make_macro (sexp ctx, sexp p, sexp e) {
   return mac;
 }
 
+sexp sexp_make_variable_transformer_op (sexp ctx, sexp self, sexp_sint_t n, sexp base_proc) {
+  sexp_assert_type(ctx, sexp_procedurep, SEXP_PROCEDURE, base_proc);
+  if (sexp_procedure_variable_transformer_p(base_proc))
+    return base_proc;
+  sexp vt_proc = sexp_alloc_type(ctx, procedure, SEXP_PROCEDURE);
+  sexp_procedure_flags(vt_proc) = (char) (sexp_uint_t) sexp_make_fixnum (sexp_unbox_fixnum(sexp_procedure_flags(base_proc)) | SEXP_PROC_VARIABLE_TRANSFORMER);
+  sexp_procedure_num_args(vt_proc) = sexp_procedure_num_args(base_proc);
+  sexp_procedure_code(vt_proc) =  sexp_procedure_code(base_proc);
+  sexp_procedure_vars(vt_proc) = sexp_procedure_vars(base_proc);
```

Comment:
I’m not entirely happy about how this copies all the fields by name, one-by-one, but I couldn’t work out how to make it work with `memcpy` (getting the right data and the right size of it behind the `sexp base_proc`)

Reply:
You don't need this - just use make-procedure.

Reply:
Oh, of course. Done eb8582f